action.yml

```diff
@@ -53,20 +53,50 @@ inputs:
 runs:
   using: 'composite'
   steps:
+    - name: Check if review is requested
+      shell: bash
+      env:
+        PR_TITLE: ${{ github.event.pull_request.title }}
+        PATTERN: \b(?:(?:no|skip)-(?:review|otter|otterai)|otter-(?:no|bye|restricted))(?:,(?:(?:no|skip)-(?:review|otter|otterai)|otter-(?:no|bye|restricted)))*\b
+      run: |
+        if echo "$PR_TITLE" | grep -Eiq "$PATTERN"; then
+          echo "🦦 No review requested, skipping code review"
+          echo "code_review_requested=false" >> $GITHUB_OUTPUT
+        else
+          echo "🔍 Code review requested"
+          echo "code_review_requested=true" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Check if PR is merged
+      shell: bash
+      env:
+        PR_STATE: ${{ github.event.pull_request.state }}
+        PATTERN: \b(?:merged|closed)\b
+      run: |
+        if echo "$PR_STATE" | grep -Eiq "$PATTERN"; then
+          echo "🦦 PR is merged, skipping code review"
+          echo "pr_merged=true" >> $GITHUB_OUTPUT
+        else
+          echo "🔍 PR is not merged, proceeding with code review"
+          echo "pr_merged=false" >> $GITHUB_OUTPUT
+        fi
     - name: Set up Python
       uses: actions/setup-python@v5
+      if: steps.check_if_review_requested.outputs.code_review_requested != 'true' || steps.check_if_pr_merged.outputs.pr_merged != 'true'
       with:
         python-version: '3.12'
         cache: 'pip'
     
     - name: Install dependencies
+      if: steps.check_if_review_requested.outputs.code_review_requested != 'true' || steps.check_if_pr_merged.outputs.pr_merged != 'true'
       shell: bash
       run: |
         python -m pip install --upgrade pip
         pip install -r ${{ github.action_path }}/requirements.txt
 
     - name: Run OtterAI Code Review
       shell: bash
+      if: steps.check_if_review_requested.outputs.code_review_requested != 'true' || steps.check_if_pr_merged.outputs.pr_merged != 'true'
```

Comment:
🔍 The conditional check for running the code review is well-placed. Ensure that the outputs are correctly set in previous steps.